base/ryu/shortest.jl

```diff
@@ -365,10 +365,10 @@ end
         c1 = (c ÷ 100) << 1
         d0 = (d % 100) << 1
         d1 = (d ÷ 100) << 1
-        memcpy(ptr, pos + olength - 2, ptr2, c0 + 1, 2)
-        memcpy(ptr, pos + olength - 4, ptr2, c1 + 1, 2)
-        memcpy(ptr, pos + olength - 6, ptr2, d0 + 1, 2)
-        memcpy(ptr, pos + olength - 8, ptr2, d1 + 1, 2)
+        memcpy(ptr, pos + olength - 2, ptr2, (c0 + 1)::UInt, 2)
```

Comment:
Only adding a type assert usually doesn't help. You tend to either have to also add a conversion or figure out why this didn't concretely infer in the first place.

Reply:
Thanks for the guidance. It appears the invalidation with `memcpy` has disappeared so I'll revert these changes. I did find some other changes I may still add in

Reply:
> It appears the invalidation with memcpy has disappeared 

Could have been https://github.com/JuliaLang/julia/pull/37802. I recall looking at this exact code for some invalidation and it is likely that I made #37802 in response to it, but I don't exactly recall.